docs/06-concepts/06-database/08-transactions.md

```diff
@@ -55,3 +55,62 @@ The available isolation levels are:
 | Serializable | `IsolationLevel.serializable` | Gives the same guarantees as `IsolationLevel.repeatableRead` but also throws if read rows are updated by other transactions. |
 
 For a detailed explanation of the different isolation levels, see the [PostgreSQL documentation](https://www.postgresql.org/docs/current/transaction-iso.html).
+
+## Savepoints
+
+Savepoints allow you to create nested transactions within a transaction. This can be useful when you want to roll back to a specific point in the transaction without rolling back the entire transaction.
+
+Read more about savepoints in the [PostgreSQL documentation](https://www.postgresql.org/docs/current/sql-savepoint.html).
+
+### Creating savepoints
+To create a savepoint, call the `createSavepoint` method on the transaction object:
+
+```dart
+await session.db.transaction((transaction) async {
+  await Company.db.insertRow(session, company, transaction: transaction);
+  // Create savepoint
+  var savepoint = await transaction.createSavepoint();
+  await Employee.db.insertRow(session, employee, transaction: transaction);
+});
+```
+
+In the example, we create a savepoint after inserting a company but before inserting the employee. This gives us the option to roll back to the savepoint and preserve the company insertion.
+
+#### Rolling back to savepoints
+
+Once a savepoint is created, you can roll back to it by calling the `rollback` method on the savepoint object:
+
+```dart
+await session.db.transaction((transaction) async {
+  // Changes preserved in the database
+  await Company.db.insertRow(session, company, transaction: transaction);
+  
+  // Create savepoint
+  var savepoint = await transaction.createSavepoint();
+
+  // Changes rolled back 
```

Comment:
Suggestion: Maybe move this comment down one line above the actual rollback command

Reply:
[ed9c853](https://github.com/serverpod/serverpod_docs/pull/223/commits/ed9c8531bc3df2c80d60020d6690ca3be16dd140)